### PR TITLE
[WIP] Fix Set for Unicode items

### PR DESCRIPTION
**Pull request by ehsantn**
As title. Adds reference counting to add/remove of Set items, and fixes function resolution for `eq`.

### CODE REVIEW

**Comment by ehsantn** [comment]
@sklam @stuartarchibald This doesn't quite work and isn't obvious why. Any ideas?

**Comment by stuartarchibald** [comment]
From `gdb`, seems like it's refcount related.

File: numba/targets/setobj.py [hunk 1]
```diff
@@ -1007,11 +1016,13 @@ def _allocate_payload(self, nentries, realloc=False):
         with builder.if_then(builder.load(ok), likely=True):
             if realloc:
                 meminfo = self._set.meminfo
-                ptr = context.nrt.meminfo_varsize_alloc(builder, meminfo,
+                ptr = context.nrt.meminfo_varsize_realloc(builder, meminfo,
```
Comment (stuartarchibald) [inline numba/targets/setobj.py]: this line is causing the problem

Comment (ehsantn) [reply numba/targets/setobj.py]: changing it to regular alloc doesn't seem to help.

Comment (stuartarchibald) [reply numba/targets/setobj.py]: hmmm, it leaked a load but it didn't segfault for me.

**Comment by ehsantn** [comment]
Closing since the right solution is implementing a typed set container.

**Pull request closed without merging**